pbxproj/pbxextensions/ProjectFiles.py

```diff
@@ -423,6 +423,89 @@ def add_folder(self, path, parent=None, excludes=None, recursive=True, create_gr
 
         return results
 
+    def add_package(self, repositoryURL, product_name, target_name, package_requirement={}):
+        package_ref, is_created = self.get_or_create_package_reference(repositoryURL, (repositoryURL, package_requirement,))
+
+        if is_created:
+            for project_object in self.objects.get_objects_in_section('PBXProject'):
+                if 'packageReferences' not in project_object:
+                    project_object['packageReferences'] = PBXList()
+
+                if package_ref.get_id() not in project_object['packageReferences']:
+                    project_object.packageReferences.append(package_ref.get_id())
```

Comment:
This seems part of the creation steps still. I think it would be better to make it part of the `get_or_create_package_reference` function, and avoid returning the `is_created` flag.

Reply:
Agreed